include/unifex/find_if.hpp

```diff
@@ -0,0 +1,205 @@
+
+namespace unifex {
+namespace _find_if {
+namespace detail {
+  template <typename Result, typename = void>
+  struct result_overload {
+    using type = type_list<Result>;
+  };
+  template <typename Result>
+  struct result_overload<Result, std::enable_if_t<std::is_void_v<Result>>> {
+    using type = type_list<>;
+  };
+}
+
+template <typename Receiver, typename Func, typename FuncPolicy>
+struct _receiver {
+  struct type;
+};
+template <typename Receiver, typename Func, typename FuncPolicy>
+using receiver_t = typename _receiver<Receiver, Func, FuncPolicy>::type;
+
+template <typename Receiver, typename Func, typename FuncPolicy>
+struct _receiver<Receiver, Func, FuncPolicy>::type {
+  UNIFEX_NO_UNIQUE_ADDRESS Func func_;
+  UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
+  UNIFEX_NO_UNIQUE_ADDRESS FuncPolicy funcPolicy_;
+
+  template<typename Iterator, typename... Values>
+  auto find_if_helper(Iterator begin_it, Iterator end_it, const Values&... values) -> Iterator {
+    // Sequential implementation
+    for(auto it = begin_it; it != end_it; ++it) {
+      if(std::invoke((Func &&) func_, *it, (Values &&) values...)) {
+        return it;
+      }
+    }
+    return end_it;
+  }
+
+  template <typename BeginIt, typename EndIt, typename... Values>
+  void set_value(BeginIt begin_it, EndIt end_it, Values&&... values) && noexcept {
+    if constexpr (noexcept(std::invoke(
+                      (Func &&) func_, *begin_it, (Values &&) values...))) {
+
+      auto result = find_if_helper(begin_it, end_it, values...);
+
+      unifex::set_value((Receiver &&) receiver_, std::move(result), (Values &&) values...);
+    } else {
+      try {
+        auto result = find_if_helper(begin_it, end_it, values...);
+
+        unifex::set_value((Receiver &&) receiver_, std::move(result), (Values &&) values...);
+      } catch (...) {
+        unifex::set_error((Receiver &&) receiver_, std::current_exception());
+      }
+    }
+  }
+
+  template <typename Error>
+  void set_error(Error&& error) && noexcept {
+    unifex::set_error((Receiver &&) receiver_, (Error &&) error);
+  }
+
+  void set_done() && noexcept {
+    unifex::set_done((Receiver &&) receiver_);
+  }
+
+  template(typename CPO, typename R)
+      (requires is_receiver_query_cpo_v<CPO> AND same_as<R, type>)
+  friend auto tag_invoke(CPO cpo, const R& r) noexcept(
+      is_nothrow_callable_v<CPO, const Receiver&>)
+      -> callable_result_t<CPO, const Receiver&> {
+    return std::move(cpo)(std::as_const(r.receiver_));
+  }
+
+  template <typename Visit>
+  friend void tag_invoke(tag_t<visit_continuations>, const type& r, Visit&& visit) {
+    std::invoke(visit, r.receiver_);
+  }
+};
+
+template <typename Predecessor, typename Func, typename FuncPolicy>
+struct _sender {
+  struct type;
+};
+template <typename Predecessor, typename Func, typename FuncPolicy>
+using sender = typename _sender<remove_cvref_t<Predecessor>, std::decay_t<Func>, FuncPolicy>::type;
```

Comment:
Generally I've been trying to avoid applying the `remove_cvref_t` and `decay_t` transformations on template args in the type-aliases here as it makes it more difficult to tell whether it's doing the correct thing at the use-site.